decidim-core/spec/lib/common_passwords_spec.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+module Decidim
+  describe CommonPasswords do
+    let(:subject) do
+      Class.new(described_class) do
+        def self.common_passwords_path
+          Rails.root.join("tmp/common-passwords.txt")
+        end
+      end
+    end
+
+    let(:organization) { create(:organization) }
+    let(:example_passwords) { %w(VJHT29061987 1234567890 q1w2e3r4t5 tooshort 0000000000) }
+    let(:urls) { Decidim::CommonPasswords::URLS }
+
+    context "when file exists and request returns body" do
+      before do
+        urls.each do |request_url|
+          stub_request(:get, request_url)
+            .with(
+              headers: { "Accept" => "*/*", "Accept-Encoding" => "gzip;q=1.0,deflate;q=0.6,identity;q=0.3", "User-Agent" => "Ruby" }
+            ).to_return(status: 200, body: example_passwords.join("\n"), headers: {})
+        end
+      end
+
+      describe "#passwords" do
+        it "returns passwords" do
+          expect(subject.instance.passwords).to be_kind_of(Array)
```

Comment:
@lahdeero I noticed another issue with this after the latest changes.

If this test is run before the `.update_passwords!` test, the password file may not exist at the overridden path and it may throw an exception.

If you run the following:
```
$ cd decidim-core
$ rm -f ../spec/decidim_dummy_app/tmp/common-passwords.txt
$ bundle exec rspec spec/lib/common_passwords_spec.rb --seed 18259
```

This would throw the following error:
```
  1) Decidim::CommonPasswords when file exists and request returns body #passwords returns passwords
     Failure/Error: raise FileNotFoundError unless File.exist?(self.class.common_passwords_path)
     
     Decidim::CommonPasswords::FileNotFoundError:
       Decidim::CommonPasswords::FileNotFoundError
     # ./lib/decidim/common_passwords.rb:16:in `initialize'
     # ./spec/lib/common_passwords_spec.rb:31:in `block (4 levels) in <module:Decidim>'
```

So, maybe we should call `subject.update_passwords!` also in this test?

Reply:
Then they would be almost identical tests. I just merged these tests.